src/extension/iSamples_resultList.js

```diff
@@ -34,42 +33,15 @@ class ResultList extends React.Component {
     this.setState({hiddenCols : updatedHiddenCols}) 
   }
 
-  // compare two arrays and checks equality 
-  arraysEqual(a, b) {
-    if (a === b) return true;
-    if (a == null || b == null) return false;
-    if (a.length !== b.length) return false;
-  
-    for (var i = 0; i < a.length; ++i) {
-      if (a[i] !== b[i]) return false;
-    }
-    return true;
-  }
-
   // prevent unnecessary rerendering
   shouldComponentUpdate(nextProps) {
+    const { results } = this.props
     this.switchView(store.getState()['query']['view']['facet']);
 
-    // compare only the necessary fields that require this component to re-render
-    const doUpdate = (nextProps.view !== this.props.view) || (nextProps.bootstrapCss !== this.props.bootstrapCss); 
-
-    // check if the cesium map point layer should be changed
-    // if it does, then invoke the child component's function  
-    const newView = store.getState()['query']['view'];
-    const fields = store.getState()['query']['searchFields'];
-    const newSearchFields = fields
-      .filter((field) => field.type !== "non-search" && field.type !== "spatialquery")
-      .map(({ collapse, hidden, ...rest }) => rest);
-
-    let pointLayerUpdate = JSON.stringify(this.state.view) !== JSON.stringify(newView) 
-                            || this.arraysEqual(this.state.searchFields, newSearchFields) ; 
-
-    if (pointLayerUpdate){
-      // redraw the cesium map point layer 
-      this.cesiumMapRef.current.redrawPointLayer(newView, newSearchFields)  
-    }
-
+    // do an update of points only when the number of points change     
+    let doUpdate = !results.pending && results.numFound !== this.state.numFound; 
```

Comment:
In the parent component, check whether we should re-render this component and the child component (`CesiumMap`) by accessing the number of documents found as result of query. If the number of documents changed, that means that we would want to re-render, which will eventually call the child component's `shouldComponentUpdate`, and redraw the point layer of the map. We don't actually need to directly access the child component through ref.